feat: Add Aiken language server support for Cardano smart contracts

BurgessTG opened the pull request:
## Summary
Add Aiken language server support to enable intelligent code assistance for Cardano smart contract development.

## Features Added
- **Auto-detection**: Automatically detects .ak files and starts Aiken LSP
- **Auto-installation**: Installs aikup and then aiken if not present via npm
- **Full LSP capabilities**: Real-time diagnostics, code completion, hover information, and error checking
- **Documentation**: Updated LSP configuration docs with Aiken examples

## How it works
1. When opencode encounters .ak files, it automatically starts the Aiken LSP server
2. If aiken is not installed, it installs aikup (the Aiken version manager) via npm
3. Then uses aikup to install the latest Aiken version
4. Starts the language server with aiken lsp command
5. Provides full LSP integration for better code assistance

## Benefits for Cardano developers
- **Catch errors early** before deployment (saves money on blockchain!)
- **Smart code completion** for Aiken functions and types  
- **Better development experience** with instant feedback
- **Seamless integration** - just open .ak files and it works

## Test plan
- [x] Verify Aiken LSP starts automatically for .ak files
- [x] Test automatic installation of aikup and aiken 
- [x] Confirm TypeScript compilation passes
- [x] Validate LSP integration follows existing patterns (TypeScript, Go, Ruby)

🤖 Generated with [opencode](https://opencode.ai)

thdxr commented:
you don't need npm! opencode itself has the functionality to run any npm package - check how we do the typescript server should work the same

BurgessTG commented:
Thank you for the feedback @thdxr! You're absolutely right - I've updated the implementation to use OpenCode's built-in npm package execution functionality.

## Changes Made:

The Aiken LSP now uses the same pattern as the TypeScript server:

```typescript
export const Aiken: Info = {
  id: "aiken",
  extensions: [".ak"],
  async spawn() {
    // Simply use bun x to run aiken lsp, just like TypeScript
    const proc = spawn(
      BunProc.which(),
      ["x", "@aiken-lang/aiken", "lsp"],
      {
        env: {
          ...process.env,
          BUN_BE_BUN: "1",
        },
      },
    )
    return {
      process: proc,
    }
  },
}
```

This approach:
- ✅ No global npm installations
- ✅ Uses OpenCode's existing infrastructure
- ✅ Matches the TypeScript server pattern exactly
- ✅ Simpler and cleaner implementation

The PR has been updated with this improved implementation. Thanks for pointing me in the right direction!

BurgessTG committed:
feat: Add Aiken language server support

- Uses OpenCode's bun x pattern for package execution
- No global installations required
- Supports .ak files for Cardano smart contract development
- Follows same pattern as TypeScript and Pyright servers
